src/snowflake/cli/plugins/nativeapp/teardown_processor.py

```diff
@@ -89,9 +97,28 @@ def drop_application(self, auto_yes: bool):
                     cc.message(f"Did not drop application object {self.app_name}.")
                     return  # The user desires to keep the app, therefore exit gracefully
 
-        # 4. All validations have passed, drop object
+        # 4. If cascade was not explicitly set, check for objects owned by the app
+        if cascade is None:
+            has_compute_pools = False  # SHOW OBJECTS OWNED BY APPLICATION
+            if has_compute_pools:
+                if interactive:
+                    cascade = typer.confirm(
+                        "There are objects owned by this app. Should they be deleted?"
+                    )
+                else:
+                    cc.warning(
+                        "There are objects owned by this app. Please explicitly set --cascade if they should be deleted, or --no-cascade otherwise."
```

Comment:
workding: set -> specify

Also, @sfc-gh-bgoel, what's the official guidance on using 'app' vs 'application' in command output?

Reply:
1. Yes, thank you for bumping this! @sfc-gh-gbloom you should use "application object" here instead of "app/application".
2. I'l like to display object names and types that are owned by the application object so the user knows the side effects of cascade (both here and line 106 like Bruno commented). Having to execute a `snow sql "show objects ..."` is an additional command we can save them the effort of typing if we just display the objects here. WDYT?

Reply:
+1 on displaying the names, I had the same comment elsewhere

Reply:
Listing the affected files now in all flows